### PR TITLE
sym constructor rewrite (594 Part 1)

### PR DESCRIPTION
**Pull request by cbm755**
I did some refactoring of @latot's commits in #692 and #594.

I've commented a bunch of things out, and removed some other commits: those things can be proposed as new PRs: I'm trying to be conservative here while still moving these changes forward.

### CODE REVIEW

File: inst/@sym/sym.m [hunk 1]
```diff
@@ -189,179 +188,224 @@
...
+  end
 
+  if (isa (x, 'char'))
+    % We now have a char; need to decide whether to use S() or Symbol() on it.
+
+    if (check)
+      % TODO: Warning if you try make a sym with the same name of a system function.
+      %symsnotfunc (x);
+
+      % TODO: tests pass without this?  Is there a example where this is needed?
```
Comment (latot) [inline inst/@sym/sym.m]: Hi, i don't know how do this test, basically, if you do this ```sym(-1)``` should not send warning, but i think we can do that in other pr, only lets hold this for the future (warning when we do ops from sym)

Thx. Cya.

Comment (cbm755) [inline inst/@sym/sym.m]: thanks, ok we'll just leave that comment for some future PR.

You ok with merging this?

File: inst/@sym/private/const_to_python_str.m [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+%% @end deftypefun
+
+function [s, flag] = const_to_python_str (x)
+
+  persistent list
+  persistent const
+
+  if (isempty (list))
+    % Table of special doubles and special strings
+    % format: number, {octave strings}, resulting python expression
```
Comment (latot) [inline inst/@sym/private/const_to_python_str.m]: if i remember right, here was a comment about always the python expression should be in the octave strings for can eval the expression in self, like ```sym(sympy(a))==a``` Are you sure you want remove that comment?

File: inst/@sym/sym.m [hunk 3]
```diff
@@ -189,179 +188,224 @@
...
+      [err flag s] = python_cmd (strrep (cmd, '{s}', x));
+
+      switch (flag{1})
+        case 1  % Bad call to python function
+          error (['Python: %s\n' ...
+                  'Error occurred using "%s" Python function, perhaps use another variable name?'],
+                 err, s);
+        case 2  % Something else
+          error (['Python: %s\n' ...
+                  'Seems you cannot use "%s" for a variable name; perhaps this is a bug?'],
```
Comment (latot) [inline inst/@sym/sym.m]: Hi!, with this text i think is possible simplify the flag structure, just to flag:
```
+             '        return (str(e), (1, "" if len(lis) == 1 else "s"), "\", \"".join(str(e) for e in lis))'
 +             '    return (str(e), (2, 0), x)' };
```
to
```
+             '        return (str(e), 1, "\", \"".join(str(e) for e in lis))'
 +             '    return (str(e), 2, x)' };
```
and change ```flag{1}``` to ```flag```

Thx. Cya.

**Comment by latot** [comment]
Hi, i think is almost ready, i put some comments, and for last, is possible keep some test for the FF issue?, like error when we try ```sym(FF(a))```.

Thx. Cya.

File: inst/@sym/private/const_to_python_str.m (hunk 2 above)
Comment (cbm755) [inline inst/@sym/private/const_to_python_str.m]: Hmm maybe I did by accident.  Not sure what comment you mean?  Was it in your #594 or in master?

**Comment by cbm755** [comment]
There is a test for `sym(FF(a))` failing.

I fixed up the the `flag` error handling, thx.

File: inst/@sym/private/const_to_python_str.m (hunk 2 above)
Comment (latot) [inline inst/@sym/private/const_to_python_str.m]: Hi, in the pr, here the comment would be like:
```
% {octave string} need contain the python expression for sympy function
```
is in https://github.com/latot/octsympy/blob/200c70e48977c244561a82fc59a19bc369803e01/inst/%40sym/private/magic_double_str.m:36

Thx. Cya.

Comment (cbm755) [inline inst/@sym/private/const_to_python_str.m]: Ok I see now.  Thanks, done.

**Comment by latot** [comment]
Hi, i think this is ready, if i found any detail i think we can send other pr.

Thx. Cya.

**Comment by cbm755** [comment]
Merging!

**Pull request merged**